From a28e4e7424a5632b1e1c368f2ec4a0b00d7eadc5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jir=CC=8Ci=CC=81=20Techet?= <techet@gmail.com>
Date: Thu, 12 Feb 2015 00:25:29 +0100
Subject: [PATCH] Make VTE independent of installation path when bundling

The wrapper shell script defines the GTK_PATH to point to
the resources directory - use this path to avoid absolute paths
in the source.
---
 src/pty.c |  7 ++++++-
 src/vte.c | 12 ++++++++++--
 2 files changed, 16 insertions(+), 3 deletions(-)

diff --git a/src/pty.c b/src/pty.c
index a4c8c07..c49e6b6 100644
--- a/src/pty.c
+++ b/src/pty.c
@@ -1181,8 +1181,13 @@ _vte_pty_start_helper(GError **error)
 		close(tunnel);
 		close(_vte_pty_helper_tunnel);
 		/* Exec our helper. */
-		execl(LIBEXECDIR "/gnome-pty-helper",
+		gchar *path = g_build_filename(g_getenv("GTK_PATH"),
+					 "libexec",
+					 "gnome-pty-helper",
+					 NULL);
+		execl(path,
 		      "gnome-pty-helper", NULL);
+		g_free(path);
 		/* Bail. */
 		_exit(1);
 	}
diff --git a/src/vte.c b/src/vte.c
index 18484a5..8b90e74 100644
--- a/src/vte.c
+++ b/src/vte.c
@@ -8272,7 +8272,10 @@ vte_terminal_set_termcap(VteTerminal *terminal, const char *path,
 	char *wpath;
 
 	if (path == NULL) {
-		wpath = g_build_filename(TERMCAPDIR,
+		wpath = g_build_filename(g_getenv("GTK_PATH"),
+					 "share",
+					 "vte",
+					 "termcap",
 					 terminal->pvt->emulation ?
 					 terminal->pvt->emulation :
 					 vte_terminal_get_default_emulation(terminal),
@@ -11720,7 +11723,12 @@ vte_terminal_class_init(VteTerminalClass *klass)
 	}
 #endif
 
-	bindtextdomain(GETTEXT_PACKAGE, LOCALEDIR);
+	gchar *localedir = g_build_filename(g_getenv("GTK_PATH"),
+				 "share",
+				 "locale",
+				 NULL);
+	bindtextdomain(GETTEXT_PACKAGE, localedir);
+	g_free(localedir);
 #ifdef HAVE_DECL_BIND_TEXTDOMAIN_CODESET
 	bind_textdomain_codeset(GETTEXT_PACKAGE, "UTF-8");
 #endif
-- 
2.2.1

